### PR TITLE
Rebuild for LLVM 14.x

### PR DESCRIPTION
**Pull request by h-vetinari**
Respin #55 for main, bump to 14.0.3

Closes #55

### CODE REVIEW

**Comment by h-vetinari** [comment]
@conda-forge-admin, please rerender

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -13,21 +13,20 @@ zip_keys:
...
-  - gxx              # [linux]
-  - vs2017           # [win]
+
 c_compiler:
   - clang_bootstrap  # [osx]
   - gcc              # [linux]
-  - vs2017           # [win]
-vc:
-  - 14
+cxx_compiler:
```
Comment (isuruf) [inline recipe/conda_build_config.yaml:20]:
```suggestion
cxx_compiler:  # [osx or linux]
```

Comment (h-vetinari) [reply recipe/conda_build_config.yaml:20]: The whole build is skipped on windows, so this is kind of cosmetic, but sure.

Comment (isuruf) [reply recipe/conda_build_config.yaml:20]: ah, yes

Comment (h-vetinari) [reply recipe/conda_build_config.yaml:20]:
```suggestion
cxx_compiler:        # [osx or linux]
```

File: recipe/conda_build_config.yaml [hunk 2]
```diff
@@ -13,21 +13,20 @@ zip_keys:
   - - macos_machine
     - cross_platform
     - uname_machine
-cxx_compiler:
-  - clang_bootstrap  # [osx]
-  - gxx              # [linux]
-  - vs2017           # [win]
+
 c_compiler:
   - clang_bootstrap  # [osx]
```
Comment (h-vetinari) [inline recipe/conda_build_config.yaml:18]:
```suggestion
c_compiler:          # [osx or linux]
  - clang_bootstrap  # [osx]
```

**Comment by h-vetinari** [comment]
I saw your second comment only after committing https://github.com/conda-forge/cctools-and-ld64-feedstock/pull/56/commits/cb9b1c9bc7b67cfef1fedc0cb6dcc17c7c231c29 - do you want me to remove it again?

Similarly, should we simply drop 12.0.1 now that the pinning has moved on to 13?

**Comment by h-vetinari** [comment]
> I saw your second comment only after committing [cb9b1c9](https://github.com/conda-forge/cctools-and-ld64-feedstock/commit/cb9b1c9bc7b67cfef1fedc0cb6dcc17c7c231c29) - do you want me to remove it again?
> 
> Similarly, should we simply drop 12.0.1 now that the pinning has moved on to 13?

I now did both of that.

**Comment by h-vetinari** [comment]
Ping @conda-forge/cctools-and-ld64, PTAL

**Comment by h-vetinari** [comment]
Gentle ping @isuruf :)

**Comment by h-vetinari** [comment]
I had been thinking - now that LLVM has moved to a much higher cadence of patch releases, should this feedstock maybe just pin to major.minor, instead of down to patch level? This would avoid rebuilding this all the time...

**Comment by h-vetinari** [comment]
Anything that I'm overlooking here...? I think this should be ready, would appreciate comments 🙃

**Comment by isuruf** [comment]
I think `major` is fine. `LLVM_LTO_LIBRARY` is build.sh has only the major version in it right?

**Comment by h-vetinari** [comment]
> I think `major` is fine. `LLVM_LTO_LIBRARY` in build.sh has only the major version in it right?

I was mostly talking about the pinning in meta.yaml & cbc.yaml, which currently goes down to patch level; I think you're right about LLVM_LTO_LIBRARY, though I'm not 100% sure what 
```
  LLVM_LTO_LIBRARY=$(find $PREFIX/lib -name "libLTO.*.*")
  LLVM_LTO_LIBRARY="$(basename $LLVM_LTO_LIBRARY)"
```
would do if the SOVERSION were to be major.minor (but non-zero minor version hasn't happened since LLVM 11 anyway).

**Comment by isuruf** [comment]
Let's do major.minor then. The only reason we need the pinning is because of the LLVM_LTO_LIBRARY being baked into the executables

**Comment by h-vetinari** [comment]
> Let's do major.minor then.

That's what I was going for as well. Should be good now?

**Comment by h-vetinari** [comment]
Should be ready now? 🙃

**Comment by isuruf** [comment]
Please reduce the number of pings or add yourself as a maintainer.

**Comment by h-vetinari** [comment]
Sorry about that, I was waiting several days between pings before, but thought that we finalized it yesterday already. I [originally](https://github.com/conda-forge/cctools-and-ld64-feedstock/pull/49#issuecomment-1030756443) didn't add myself as maintainer because I feel I don't understand this feedstock well enough, but will do so now to bother you less.

**Comment by h-vetinari** [comment]
@conda-forge-admin, please rerender

**Comment by isuruf** [comment]
Thanks

**Pull request merged**